Hermes/appscale/hermes/stats/producers/cluster_stats.py

```diff
@@ -16,280 +16,110 @@
-                                    last_utc_timestamp, limit, include_lists,
-                                    fetch_latest_only):
-  # Security header
-  headers = {SECRET_HEADER: options.secret}
-  # Build query arguments
-  arguments = {}
-  if last_utc_timestamp is not None:
-    arguments['last_utc_timestamp'] = last_utc_timestamp
-  if limit is not None:
-    arguments['limit'] = limit
-  if include_lists is not None:
-    arguments['include_lists'] = include_lists.asdict()
-  if fetch_latest_only:
-    arguments['fetch_latest_only'] = True
-
-  url = "http://{ip}:{port}/{path}".format(
-    ip=node_ip, port=constants.HERMES_PORT, path=method_path)
-  request = httpclient.HTTPRequest(
-    url=url, method='GET', body=json.dumps(arguments), headers=headers,
-    validate_cert=False, request_timeout=REQUEST_TIMEOUT,
-    allow_nonstandard_methods=True
-  )
-  async_client = httpclient.AsyncHTTPClient()
-
-  try:
-    # Send Future object to coroutine and suspend till result is ready
-    response = yield async_client.fetch(request)
-  except HTTPError as e:
-    logging.error("Failed to get stats from {} ({})".format(url, e))
-    raise gen.Return([])
-
-  try:
-    stats_dictionaries = json.loads(response.body)
-    snapshots = [
-      converter.stats_from_dict(stats_class, stats_dict)
-      for stats_dict in stats_dictionaries
-    ]
-  except TypeError as err:
-    msg = u"Can't parse stats snapshot ({})".format(err)
-    raise BadStatsListFormat(msg), None, sys.exc_info()[2]
-
-  raise gen.Return(snapshots)
+  @gen.coroutine
+  def _fetch_remote_stats_async(self, node_ip, newer_than, include_lists):
+    # Security header
+    headers = {SECRET_HEADER: options.secret}
+    # Build query arguments
+    arguments = {}
+    if include_lists is not None:
+      arguments['include_lists'] = include_lists.asdict()
+    if newer_than:
+      arguments['newer_than'] = newer_than
+
+    url = "http://{ip}:{port}/{path}".format(
+      ip=node_ip, port=constants.HERMES_PORT, path=self.method_path)
+    request = httpclient.HTTPRequest(
+      url=url, method='GET', body=json.dumps(arguments), headers=headers,
+      validate_cert=False, request_timeout=REQUEST_TIMEOUT,
+      allow_nonstandard_methods=True
+    )
+    async_client = httpclient.AsyncHTTPClient()
+
+    try:
+      # Send Future object to coroutine and suspend till result is ready
+      response = yield async_client.fetch(request)
+    except HTTPError as e:
+      logging.error("Failed to get stats from {} ({})".format(url, e))
+      raise gen.Return(None)
+
+    try:
+      snapshot = json.loads(response.body)
+      raise gen.Return(converter.stats_from_dict(self.stats_model, snapshot))
+    except TypeError as err:
+      msg = u"Can't parse stats snapshot ({})".format(err)
+      raise BadStatsListFormat(msg), None, sys.exc_info()[2]
+
+
+class ClusterNodesStatsSource(ClusterStatsSource):
+  ips_getter = staticmethod(appscale_info.get_all_ips)
```

Comment:
maybe.. I decided to use this name, because 
`get_ips = staticmethod(appscale_info.get_all_ips)` looks worse to me.
I was considering ips_getter as something thirdparty - an attribute of stats source.